Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp

```diff
@@ -527,6 +527,19 @@ void FunctionDefinitionWriter::emitNecessaryHelpers()
         m_stringBuilder.append(m_indent, "}\n\n"_s);
     }
 
+    if (m_shaderModule.usesFtoi()) {
+        m_stringBuilder.append(m_indent, "template <typename T, typename S>\n"_s,
+            m_indent, "static T __wgslFtoi(S value)\n"_s,
+            m_indent, "{\n"_s);
+        {
+            IndentationScope scope(m_indent);
+            m_stringBuilder.append(m_indent, "auto min = metal::numeric_limits<T>::min();\n"_s,
+                m_indent, "auto max = metal::numeric_limits<T>::max();\n"_s,
```

Comment:
Do we need to special case `__fp16`?

E.g., similar to the logic from `std::optional<To> convertFloat(From value)` in `ConstantValue.h`?
```
    if constexpr (std::is_floating_point<To>::value) {
        max = std::numeric_limits<To>::max();
        lowest = std::numeric_limits<To>::lowest();
    } else {
        max = 0x1.ffcp15;
        lowest = -max;
    }
```

Reply:
but `min` is different in this case, here's max / min for reference for `__fp16`:
```
#define  HALF_MAX  0x1.ffcp15
#define  HALF_MIN  0x1.0p-14
```